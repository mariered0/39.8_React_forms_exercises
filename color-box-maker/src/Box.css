.Box {
    margin: 10px;
    display: inline-block;

}

.delete-btn {
    margin: 10px;
    width: 20px;
    height: 20px;
}