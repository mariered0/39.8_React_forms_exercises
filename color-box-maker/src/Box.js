import React from "react";
import "./Box.css";

const Box = ({ id, width, height, color, removeBox }) => {
  const handleDelete = (e) => {
    removeBox(e.target.getAttribute("id"));
  };
  return (
    <div className="Box-container">
      <div
        className="Box"
        id={id}
        style={{
          width,
          height,
          backgroundColor: color,
        }}
      ></div>
      <button className="delete-btn" onClick={handleDelete}  id={id} >
        x
      </button>
    </div>
  );
};

export default Box;
