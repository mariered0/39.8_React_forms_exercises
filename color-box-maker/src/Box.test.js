import React from "react";
import { render, queryByText } from "@testing-library/react";
import Box from "./Box";


it("renders without crashing", function() {
    render(<Box />);
});

it("matches snapshot", function() {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});
