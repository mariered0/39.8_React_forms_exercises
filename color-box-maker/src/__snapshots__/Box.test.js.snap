// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
    />
    <button
      class="delete-btn"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;
